### ISSUE TITLE
Buck project fails to create a intellij project

### ISSUE THREAD

**Issue by minid33** [facebook/buck#241]
After commit f3b6cd48bab0247e8bf85ef3c52afd9b7e26dd95q buck fails to create an intellij project when using `buck project`.

```
Traceback (most recent call last):
  File "/Users/jaimie/git/buck/.buckd/tmp/buck_run.b6RWC_/buck7374376320008866097.py", line 1271, in <module>
    main()
  File "/Users/jaimie/git/buck/.buckd/tmp/buck_run.b6RWC_/buck7374376320008866097.py", line 474, in main
    values = buildFileProcessor.process(build_file.rstrip())
  File "/Users/jaimie/git/buck/.buckd/tmp/buck_run.b6RWC_/buck7374376320008866097.py", line 402, in process
    implicit_includes=self._implicit_includes)
  File "/Users/jaimie/git/buck/.buckd/tmp/buck_run.b6RWC_/buck7374376320008866097.py", line 385, in _process_build_file
    len_suffix = -len('/' + self._build_file_name)
TypeError: cannot concatenate 'str' and 'NoneType' objects
[2015-02-03 11:06:44.677][error][command:0f5d7877-5126-4efe-92a7-0db87aa6f4e0][tid:33][com.facebook.buck.json.ProjectBuildFileParser] Process java.lang.UNIXProcess@1c3288ba exited with error code 1
BUILD FAILED: Parse error for BUCK file /Users/jaimie/git/buck/assets/android/BUCK: End of input at line 1 column 1
```

You can test this by pulling head of master from the buck project and using `buck project`.

**Comment by sdwilsh** [comment]
I stand corrected.  The call to `buck.py` is missing `--build_file_name`, which is highly suspicious.

**Comment by Coneko** [comment]
Can you try running `buckd` to restart Buck and see if you still get the same error or a different one?

**Comment by sdwilsh** [comment]
I'm still seeing it even after restarting

**Comment by Coneko** [comment]
Was this fixed?

**Comment by FrancisToth** [comment]
I think this is fixed. However, the generated IDEA's files are pretty bad. If we take the example of .idea/misc.xml : 
before <code>buck project</code> :
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="ProjectModuleManager">
    <modules>
      <module fileurl="file://$PROJECT_DIR$/third-party/java/aosp/aosp.iml" filepath="$PROJECT_DIR$/third-party/java/aosp/aosp.iml" />
      <module fileurl="file://$PROJECT_DIR$/buck.iml" filepath="$PROJECT_DIR$/buck.iml" />
      <module fileurl="file://$PROJECT_DIR$/src/com/facebook/buck/maven/maven.iml" filepath="$PROJECT_DIR$/src/com/facebook/buck/maven/maven.iml" />
      <module fileurl="file://$PROJECT_DIR$/scripts/scripts.iml" filepath="$PROJECT_DIR$/scripts/scripts.iml" />
      <module fileurl="file://$PROJECT_DIR$/src/com/facebook/buck/junit/testrunner.iml" filepath="$PROJECT_DIR$/src/com/facebook/buck/junit/testrunner.iml" />
    </modules>
  </component>
</project>
```

after <code>buck project</code>  : 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="EntryPointsManager">
    <entry_points version="2.0" />
  </component>
  <component name="FrameworkDetectionExcludesConfiguration">
    <type id="android" />
  </component>
  <component name="IdProvider" IDEtalkID="0DE2DAD55CCA0E8D48163E3F370D8939" />
  <component name="ProjectResources">
    <default-html-doctype>http://www.w3.org/1999/xhtml</default-html-doctype>
  </component>
  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" assert-keyword="true" jdk-15="true" project-jdk-name="1.7" project-jdk-type="JavaSDK">
    <output url="file://$PROJECT_DIR$/build-ij/classes" />
  </component>
</project>
```

**Comment by Coneko** [comment]
`buck project` doesn't generate a `misc.xml` file, the first file you posted looks like a `modules.xml`.

**Comment by FrancisToth** [comment]
Sorry my bad, I looked at the wrong file (this is what happens when you work too late :)). However, there is a clear difference (just try to open IDEA before and after). I'll try to look into that this week.

Sorry again .

**Comment by sdwilsh** [comment]
Martin is working on revamping our project generation.  It's hidden behind an experimental flag right now, but it should be much better than what we have today.

**Comment by marcinkosiba** [comment]
Actually, `buck project` recently started generating a misc.xml file.
The experimental flag is '--experimental-ij-generation' and that's what I'm concentrating on right now. I'd appreciate it if you could check it out. Depending on the input the generated project may heavily rely on buck-out being up to date, so it may not be too great for you if 'buck build' is not part of your usual workflow.

**Comment by sdwilsh** [comment]
@FrancisToth, have you had a chance to try out the new project behavior?

**Comment by strangemonad** [comment]
@sdwilsh @marcinkosiba I've just recently ported a large-ish project from maven to buck. Everything is now working (mostly) except for project generation. The experimental flag didn't seem to fix it but I was sure how to debug further

**Comment by nsharma-git** [comment]
I'm a beginner trying to import the gerrit project in intelliJ. I tried both "--ide intellij" and "--experimental-ij-generation"
gerrit/gerrit-acceptance-framework/BUCK", line 54, in <module>
    java_sources(
NameError: name 'java_sources' is not defined

Buck Version - v2016.03.28.01
Gerrit - 2.12.2-stable
Python - Python 2.7.10
OS - Mac OS 10.10

**Comment by sdwilsh** [comment]
@strangemonad - @alsutton has done a lot of work on this an the project flow is now vastly improved.

@nsharma-git - that's probably worth raising with the gerrit project.  That's a special rule that they have and is unrelated to Buck.

**Comment by mkillianey** [comment]
Should be fixed in dabe44eac57c296cdf87ede3f210935f8ae8debb.
